### PR TITLE
Add an error message

### PR DESCRIPTION
**Pull request by ASkrypka**
Extending all Guard methods by allowing add an error message

### CODE REVIEW

File: src/GuardClauses/GuardClauseExtensions.cs [hunk 1]
```diff
@@ -558,13 +635,19 @@ public static TimeSpan NegativeOrZero([JetBrainsNotNull] this IGuardClause guard
...
         {
             if (!Enum.IsDefined(typeof(T), input))
             {
-                throw new InvalidEnumArgumentException(parameterName, Convert.ToInt32(input), typeof(T));
+                if (message == "")
+                {
+                    throw new InvalidEnumArgumentException(parameterName, Convert.ToInt32(input), typeof(T));
+                }
+                throw new InvalidEnumArgumentException(message, new InvalidEnumArgumentException(parameterName, Convert.ToInt32(input), typeof(T)));
+
```
Comment (apm1grb) [inline src/GuardClauses/GuardClauseExtensions.cs]: wrong spacing: empty line after throw, instead before

File: src/GuardClauses/GuardClauseExtensions.cs [hunk 2]
```diff
@@ -577,13 +660,19 @@ public static TimeSpan NegativeOrZero([JetBrainsNotNull] this IGuardClause guard
...
+        public static T OutOfRange<T>([JetBrainsNotNull] this IGuardClause guardClause, T input, [JetBrainsNotNull] string parameterName, string message = "") where T : struct, Enum
         {
             if (!Enum.IsDefined(typeof(T), input))
             {
-                throw new InvalidEnumArgumentException(parameterName, Convert.ToInt32(input), typeof(T));
+                if (message == "")
+                {
+                    throw new InvalidEnumArgumentException(parameterName, Convert.ToInt32(input), typeof(T));
+                }
+
```
Comment (apm1grb) [inline src/GuardClauses/GuardClauseExtensions.cs]: consistency of spacing: sometimes with an empty line, sometimes without

I would recommend with an empty line after the closing brace of the if (I assume that is today the default spacing rule)

**Review by apm1grb — CHANGES_REQUESTED** [review]
With making the spacing consistent for my it looks fine.

**Review by apm1grb — APPROVED** [review]
Try next time to have such changes decoupled from the "real" work. 
From the packaging perspective, it was wrong to have it in the last commit.

File: src/GuardClauses/GuardClauseExtensions.cs (hunk 1 above)
Comment (ASkrypka) [reply src/GuardClauses/GuardClauseExtensions.cs]: ...

File: src/GuardClauses/GuardClauseExtensions.cs (hunk 2 above)
Comment (ASkrypka) [reply src/GuardClauses/GuardClauseExtensions.cs]: ...

**Pull request merged**